src/components/useTodoItems.js

```diff
@@ -0,0 +1,274 @@
+      }
+    }
+  `;
+  const [ addOne, ] = useMutation(ADD_ONE);
+  const addTodo = async task => {
+    const { data } = await addOne(
+      {
+        variables: {
+          "data": {
+            "task": task,
+            "owner_id": userId
+          }
+        }
+      }
+    );
+    const todo = { task, owner_id: userId };
+    dispatch({ type: "addTodo", payload: { ...todo, _id: data.insertOneItem._id } });
+  };
+
+  const DELETE_ONE = gql`
+    mutation($query: ItemQueryInput!) {
+      deleteOneItem(query: $query) {
+        _id
+      }
+    }
+  `;
+  const [deleteOne, ] = useMutation(DELETE_ONE);
+  const removeTodo = async todoId => {
+    await deleteOne(
+      {
+        variables: {
+          "data": {
+            "_id": todoId,
+            "owner_id": userId
+          }
+        }
+      }
+    );
+    dispatch({ type: "removeTodo", payload: { id: todoId } });
+  };
+
+  const DELETE_ALL = gql`
+    mutation {
+      deleteManyItems {
+        deletedCount
+      }
+    }
+  `;
+  const [deleteAll, ] = useMutation(DELETE_ALL);
+  const clearTodos = async () => {
+    await deleteAll();
+    dispatch({ type: "clearTodos" });
+  };
+
+  const DELETE_COMPLETED = gql`
+    mutation($query:ItemQueryInput) {
+      deleteManyItems(query: $query) {
+        deletedCount
+      }
+    }
+  `;
+  const [deleteCompleted, ] = useMutation(DELETE_COMPLETED);
+  const clearCompletedTodos = async () => {
+    await deleteCompleted(
+      {
+        variables: {
+          "query": {
+            "checked": true,
+            "owner_id": userId
+          }
+        }
+      }
+    );
+    dispatch({ type: "clearCompletedTodos" });
+  };
+
+  const SET_STATUS = gql`
+    mutation($query: ItemQueryInput!, $set: ItemUpdateInput!) {
+      updateOneItem(query: $query, set: $set) {
+        _id
+      }
+    }
+  `;
+  const [setStatus, ] = useMutation(SET_STATUS);
+  const setTodoCompletionStatus = async (todoId, status) => {
+    await setStatus(
+      { 
+        variables: {
+          "query": { "_id": todoId, "owner_id": userId },
+          "set": { "checked": status }
+        }
+      }
+    );
+    dispatch({ type: "setTodoStatus", payload: { todoId, status } });
+  };
+
+  const COMPLETE_ALL = gql`
+    mutation($query: ItemQueryInput, $set: ItemUpdateInput!) {
+      updateManyItems(query: $query, set: $set) {
+        modifiedCount
+      }
+    }
+  `;
+  const [completeAll, ] = useMutation(COMPLETE_ALL);
+  const completeAllTodos = async () => {
+    await completeAll(
+      { 
+        variables: {
+          "query": { "owner_id": userId },
+          "set": { "checked": true }
+        }
+      }
+    );
+    dispatch({ type: "completeAllTodos" });
+  };
+
+  const FIND_TODO = gql`
+    query($query: ItemQueryInput!) {
+      item(query: $query) {
+        checked
+        _id
+        task
+      }
+    }
+  `;
+  const [findTodo, { data: toggleData }]
+    = useLazyQuery(FIND_TODO, { fetchPolicy: "network-only"});
```

Comment:
What's the `fetchPolicy` for here?

Reply:
Toggling wasn't working because this request was using cached values instead of making network requests which had stale values for the `checked` field. This option forces the query to do a network request instead of using the cache.